### PR TITLE
Upgraded eslint to major version 8, added TypeScript config for node14

### PR DESCRIPTION
**Pull request by attilakiss84**
- Breaking: eslint major version updated to 8
- Dependencies moved to peerDependencies
- Added basic TypeScript configuration for node14

### CODE REVIEW

File: node14-typescript.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        'node/no-unsupported-features/es-syntax': 'off',
+        'no-array-constructor': 'off', // must be disabled so the typescript one can take over.
+        'no-dupe-keys': 'error',
+        'no-extra-semi': 'off', // must be disabled so the typescript one can take over.
+        'no-restricted-imports': [
+          'error',
+          {
+            patterns: ['*/config']
+          }
+        ],
```
Comment (v-hal) [inline node14-typescript.js:42]: We might want to omit this. It's likely a project-specific configuration.

**Review by v-hal — APPROVED** [review]
Looks great, thanks for adding. I commented about some minor changes.

File: node14-typescript.js (hunk 1 above)
Comment (attilakiss84) [reply node14-typescript.js]: Good point, removed

File: node14-typescript.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        'sort-keys': ['error', 'asc', { caseSensitive: false, minKeys: 5 }],
+        'spaced-comment': 'error',
+        '@typescript-eslint/adjacent-overload-signatures': 'error',
+        '@typescript-eslint/ban-ts-comment': 'error',
+        '@typescript-eslint/ban-types': 'error',
+        '@typescript-eslint/consistent-type-assertions': 'error',
+        '@typescript-eslint/explicit-function-return-type': 'error',
+        '@typescript-eslint/member-delimiter-style': 'error',
+        '@typescript-eslint/no-array-constructor': 'error',
+        // 'no-empty-function': 'off',
```
[the remark this answers is not in the document]
Comment (attilakiss84) [reply node14-typescript.js]: Removed

**Pull request merged**